src/core/algorithms/md/hymd/hymd.cpp

```diff
@@ -175,25 +175,40 @@ unsigned long long HyMD::ExecuteInternal() {
 void HyMD::RegisterResults(SimilarityData const& similarity_data,
                            std::vector<lattice::MdLatticeNodeInfo> lattice_mds) {
     std::size_t const column_match_number = similarity_data.GetColumnMatchNumber();
-    std::vector<model::md::ColumnMatch> column_matches;
-    column_matches.reserve(column_match_number);
+    std::size_t const trivial_column_match_number = similarity_data.GetTrivialColumnMatchNumber();
+    std::size_t const all_column_match_number = column_match_number + trivial_column_match_number;
+    auto const& sorted_to_original = similarity_data.GetIndexMapping();
+    std::vector<model::md::ColumnMatch> column_matches =
+            std::vector<model::md::ColumnMatch>(all_column_match_number);
     for (Index column_match_index = 0; column_match_index < column_match_number;
          ++column_match_index) {
         auto [left_col_index, right_col_index] =
                 similarity_data.GetColMatchIndices(column_match_index);
-        column_matches.emplace_back(
+        column_matches[sorted_to_original[column_match_index + trivial_column_match_number]] = {
```

Comment:
> `column_match_index + trivial_column_match_number`

This seems to rely on the column matches to be sorted on the LHS decision boundary number. I want to experiment with different sortations, so please don't rely on that.